test(docs-infra): ignore empty lines in `URLS_TO_REDIRECT.txt`

gkalpak opened the pull request:
Previously, the `URLS_TO_REDIRECT.txt` file was expected to not contain any empty lines. This could easily result in test errors when updating the file, since it is common for IDEs/editors to automatically ensure there is an empty line at the end of a saved file ([example failure][1]).

This commit updates the test helpers to be able to cope with empty or whitespace-only lines in `URLS_TO_REDIRECT.txt` by ignoring such lines.

[1]: https://circleci.com/gh/angular/angular/965534

gkalpak committed:
test(docs-infra): ignore empty lines in `URLS_TO_REDIRECT.txt`

Previously, the `URLS_TO_REDIRECT.txt` file was expected to not contain
any empty lines. This could easily result in test errors when updating
the file, since it is common for IDEs/editors to automatically ensure
there is an empty line at the end of a saved file
([example failure][1]).

This commit updates the test helpers to be able to cope with empty or
whitespace-only lines in `URLS_TO_REDIRECT.txt` by ignoring such lines.

[1]: https://circleci.com/gh/angular/angular/965534

mary-poppins commented:
You can preview 93e0eb6 at https://pr41680-93e0eb6.ngbuilds.io/.

petebacondarwin commented on aio/tests/deployment/shared/helpers.ts:
```diff
@@ -41,7 +41,10 @@ export function loadRedirects(): FirebaseRedirectConfig[] {
+  const urls = readFileSync(pathToLegacyUrls, 'utf8')
+      .split('\n')
+      .filter(line => line.trim() !== '')
+      .map(line => line.split('\t'));
```
I wonder if we should also change the delimiter to something non-whitespacy, since the tab character is hard to spot in the file? 🤔

gkalpak replied:
Happy to change that too. We could switch to either any whitespace (since the URLs should not contain unencoded whitespace) or something like ` --> ` (to hint about the "from" --> "to" relationship of the links).
(Also happy to go with a different separator.)

WDYT?

petebacondarwin replied:
I like `-->`

gkalpak replied:
I added a new commit to switch to ` --> `. PTAL

gkalpak committed:
test(docs-infra): change URL separator from `\t` to `-->` in `URLS_TO_REDIRECT.txt`

Previously, the `URLS_TO_REDIRECT.txt` file was expected to separate
URLs by a `\t` character. This could easily been mistaken for a regular
space and would often result in test errors when updating the file.

This commit updates the `URLS_TO_REDIRECT.txt` file and the
corresponding test helpers to use `-->` (potentially surrounded by
whitespace) as a separator for the URLs. This also gives a hint about
the "from" --> "to" relationship of the redirect URL pair.

mary-poppins commented:
You can preview 394e07c at https://pr41680-394e07c.ngbuilds.io/.

petebacondarwin reviewed (approved):
Super!
